### PR TITLE
improve diff abund api

### PR DESCRIPTION
**Pull request by d-callan**
depends on https://github.com/microbiomeDB/microbiomeComputations/pull/20

this does a couple things:
1. introduces a wrapper around microbiomeComputations::differentialAbundance. this wrapper, rather than expecting a Comparator, takes a covariate name and optional predicates for defining groups within that covariate.
2. adds a wrapper around Maaslin2::Maaslin2 which takes a CollectionWithMetadata object, grabs out of that object what it needs in terms of data, and otherwise just passes along whatever arguments maaslin would usually take.
3. adds a new method DESeqDataSetFromCollection which will take an AbsoluteAbundanceData object and build a DESeqDataSet from it. this can then be used w deseq as one usually would. the method is named as it is in order to align as much as possible w the deseq api, where we already see things like DESeqDataSetFromMatrix.

im considering that getCollection maybe should check to see if all values are ints, and in that case return AbsoluteAbundanceData rather than AbundanceData objects. id appreciate thoughts and opinions there, bc i dont generally like conditionally returning different things from the same method. maybe an arg to override that behavior if desired for some reason? or an arg to turn it on?

### CODE REVIEW

File: tests/testthat/test-differentialAbundance.R [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        method='Maaslin2', 
+        verbose=FALSE
+    )
+    expect_equal(inherits(diffAbundOutput, "ComputeResult"), TRUE)
+    expect_equal(inherits(diffAbundOutput@statistics, "DifferentialAbundanceResult"), TRUE)
+    expect_equal(inherits(diffAbundOutput@statistics@statistics, "data.frame"), TRUE)
+    expect_equal(nrow(diffAbundOutput@statistics@statistics) > 0, TRUE)
+
+    # building predicates on continuous vars works
+    diffAbundOutput <- MicrobiomeDB::differentialAbundance(
```
Comment (asizemore) [inline tests/testthat/test-differentialAbundance.R:69]: this is an awesome example. Could put in the function description

**Review by asizemore — COMMENTED** [review]
Looks great! The new api is helpful. I'd also have dan take a look to see what he thinks about the predicate part. I've seen examples like `diffabund(data, comparator = "hairColor", groupA = ['red', 'blue'], groupB = ['green'])` that are perhaps more familiar to people? Regardless what you have here is more powerful, but it's possible we add another version that's more similar to other methods if users (dan) think that's helpful. And of course my usage of simlar functions previously may not at all be what the mbio folks do these days!

**Comment by d-callan** [comment]
> diffabund(data, comparator = "hairColor", groupA = ['red', 'blue'], groupB = ['green'])
i wanted to be able to handle continuous vars as well w a single api, and this is where i landed. looking at other tools, there doesnt appear to be any consistency in api for covariates that i can see either.. though ill say using formulas is something deseq supports, and i suppose i was hoping predicates would be no more confusing than that..

File: R/method-differentialAbundance.R [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+    verbose <- veupathUtils::matchArg(verbose)
+    
+    if (data.table::uniqueN(data@sampleMetadata@data[[covariate]]) < 2) {
+        stop("Argument 'covariate' must have at least two unique values")
+    }
+
+    data@sampleMetadata@data[[covariate]] <- assignGroups(data@sampleMetadata@data[[covariate]], groupAPredicate, groupBPredicate)
+    comparator <- buildBinaryComparator(covariate, 'groupA', 'groupB')
+
+    ## microbiomeComputations will remove for us rows not in either group, and provide validation
```
[the remark this answers is not in the document]
Comment (d-callan) [reply R/method-differentialAbundance.R:129]: fairly certain it does

File: R/method-differentialAbundance.R [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+    return(microbiomeComputations::differentialAbundance(data, comparator = comparator, method = method, verbose = verbose))
+})
+
+#### NOTE: While i think its important for people to be able to recreate the computes from the site, and tried to make
+#### that easier w the differentialAbundance wrapper method above, I also recognize that our
+#### microbiomeComputations implementation is limited. So these are methods that should let people pass our
+#### objects more-or-less directly to Maaslin2 and DESeq2, using all the same arguments they would usually be able to.
+
+#' Maaslin2 from a Collection
```
[the remark this answers is not in the document]
Comment (d-callan) [reply R/method-differentialAbundance.R:138]: i considered possible alternative ways of factoring this, and think theres room for improvement, but ultimately decided this was good enough to get us going and we can make a separate pr refactoring in the future if needed. it wasnt quite straightforward what exactly the relationship should be.

File: R/method-differentialAbundance.R [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+    comparator <- buildBinaryComparator(covariate, 'groupA', 'groupB')
+
+    ## microbiomeComputations will remove for us rows not in either group, and provide validation
+    return(microbiomeComputations::differentialAbundance(data, comparator = comparator, method = method, verbose = verbose))
+})
+
+#### NOTE: While i think its important for people to be able to recreate the computes from the site, and tried to make
+#### that easier w the differentialAbundance wrapper method above, I also recognize that our
+#### microbiomeComputations implementation is limited. So these are methods that should let people pass our
+#### objects more-or-less directly to Maaslin2 and DESeq2, using all the same arguments they would usually be able to.
```
[the remark this answers is not in the document]
Comment (d-callan) [reply R/method-differentialAbundance.R:136]: the function descriptions for maaslin and deseqdataset are nearly directly copied from the original functions. idk how far from that id like to stray..

**Pull request merged**